app/controllers/movies_controller.rb

```diff
@@ -1,5 +1,7 @@
 class MoviesController < ApplicationController
   def index
-    @movies = Movie.where(genre: ["basic", "git", "ruby", "rails"]).order(id: :asc).page(params[:page]).per(9)
+    genre_movies = Movie.where(genre: ["basic", "git", "ruby", "rails"]).order(id: :asc).page(params[:page]).per(9)
+    @q = genre_movies.ransack(params[:q])
+    @movies = @q.result.page(params[:page]).per(9)
```

Comment:
ページネーションは最後に反映しなければなりません。この形式ではありませんか。

```suggestion
    @q = Movie.where(genre: ["basic", "git", "ruby", "rails"]).ransack(params[:q])
    @movies = @q.result.order(id: :asc).page(params[:page]).per(9)
```

テキスト教材も同様に修正しましょう。

Reply:
指摘頂いた通りページネーションを最後にするように変更致しました。

Reply:
テキスト教材もページネーションを修正致しました。